mpl_interactions/generic.py

```diff
@@ -599,11 +607,15 @@ def hyperslicer(
 
     names = None
     axes = None
-    if "names" in kwargs:
-        names = kwargs.pop("names")
+    if arr_type != "xarray":
+        if "names" in kwargs:
+            names = kwargs.pop("names")
+
+        elif "axes" in kwargs:
+            axes = kwargs.pop("axes")
```

Comment:
I think names would get overridden. Since everything sort of funnels into creating `axes` so when i `get_hs_axes` before the big conditional mess, no matter what other kwargs you have, the axes from the xarray are the thing that get processed by the conditional mess zone.

Reply:
But the idea here is definitely that if you want to pass args to hyperslicer use numpy/dask arrays. Otherwise put that same info into an xarray and let this thing read the xarray. maybe theres a better way? Also this is the sort of reason that this functionality could really benefit from stress testing by an advanced xarray user.